CMakeLists.txt

```diff
@@ -167,37 +147,32 @@ if (NOT BUILD_TESTS_ONLY)
   #############################################################################
   # LINKING AND INCLUDE DIRECTORIES
   #############################################################################
+  target_compile_options(
+    ${PROJECT_NAME}
+    PUBLIC
+      ${offload_flags}
```

Comment:
Putting this in PUBLIC exports it to `find_package(rocshmem)` consumers that use `*_link_libraries(roc::rocshmem)`; this will cause an error if they provide a different set of `--offload-target` that is incompabible (good), this also forces them to compile all variants we compiled in the library (bad), keep?